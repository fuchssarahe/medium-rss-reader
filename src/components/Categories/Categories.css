.Categories {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.Categories__category {
  background-color: #F4F4ED;
  border: 1px solid black;
  border-radius: 0.25em;
  margin: 0.5em;
  padding: 0.25em;
  box-shadow: 0.125em 0.125em #808782;
}

.Categories__category:hover {
  background-color: #808782;
  color: #F4F4ED;
  cursor: pointer;
}
