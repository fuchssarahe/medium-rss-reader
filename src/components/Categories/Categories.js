import React from 'react';
import './Categories.css';

const Categories = ({ categories = [], removable = false, onClick = null }) => {
  return (
    <h4 className='Categories'>
      {
        categories.map(category => (
          <button className='Categories__category' key={category} onClick={() => onClick(category)}>
            {category}
            { removable && <span> x</span> }
          </button>)
        )
      }
    </h4>
  );
}

export default Categories;
