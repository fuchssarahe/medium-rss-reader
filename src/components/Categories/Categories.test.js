import React from 'react';
import { shallow } from 'enzyme';
import Categories from './Categories';


describe('<Categories />', () => {
  it('renders categories as individual elements', () => {
    expect(shallow(
      <Categories
        categories={['a', 'b', 'c']}
        removable={false}
      />
    )).toMatchSnapshot();
  })

  describe('when "categories" is undefined', () => {
    test('it renders an empty element', () => {
      expect(shallow(<Categories />)).toMatchSnapshot();
    })
  });

  describe('when no categories are present', () => {
    test('it renders an empty element', () => {
      expect(shallow(<Categories categories={[]} />)).toMatchSnapshot();
    })
  });

  describe('when categories are removable', () => {
    test('it shows an X on category', () => {
      expect(shallow(<Categories categories={['a']} removable={true} />)).toMatchSnapshot();
    })
  });

  describe('when the category can trigger an action', () => {
    test('it calls the provided handler on click', () => {
      const actionSpy = jest.fn();
      const component = shallow(<Categories categories={['a']} onClick={actionSpy} />);
      component.find('.Categories__category').first().simulate('click');
      expect(actionSpy).toHaveBeenCalled();
    });
  })
});
