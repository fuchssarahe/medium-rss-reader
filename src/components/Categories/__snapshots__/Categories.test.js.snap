// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders categories as individual elements 1`] = `
<h4
  className="Categories"
>
  <button
    className="Categories__category"
    key="a"
    onClick={[Function]}
  >
    a
  </button>
  <button
    className="Categories__category"
    key="b"
    onClick={[Function]}
  >
    b
  </button>
  <button
    className="Categories__category"
    key="c"
    onClick={[Function]}
  >
    c
  </button>
</h4>
`;

exports[`<Categories /> when "categories" is undefined it renders an empty element 1`] = `
<h4
  className="Categories"
/>
`;

exports[`<Categories /> when categories are removable it shows an X on category 1`] = `
<h4
  className="Categories"
>
  <button
    className="Categories__category"
    key="a"
    onClick={[Function]}
  >
    a
    <span>
       x
    </span>
  </button>
</h4>
`;

exports[`<Categories /> when no categories are present it renders an empty element 1`] = `
<h4
  className="Categories"
/>
`;
